odoo/tools/misc.py

```diff
@@ -959,23 +959,26 @@ def update(self, *args, **kwargs):
     def __hash__(self):
         return hash(frozenset((key, freehash(val)) for key, val in self.items()))
 
-class Collector(Mapping):
+
+class Collector(dict):
     """ A mapping from keys to lists. This is essentially a space optimization
         for ``defaultdict(list)``.
     """
-    __slots__ = ['_map']
-    def __init__(self):
-        self._map = {}
+    __slots__ = ()
+
+    def __getitem__(self, key):
+        return self.get(key, ())
+
+    def __setitem__(self, key, val):
+        if val:
+            super().__setitem__(key, val)
+        else:
+            super().pop(key, None)
```

Comment:
That seems pretty dodgy, and really doesn't match the docstring.

Reply:
I changed the docstring ;-)